packages/connect/src/utils/pathUtils.ts

```diff
@@ -207,16 +209,18 @@ export const fixPath = <
 >(
     utxo: T,
 ): ProtoWithAddressN<T> => {
+    const newUtxo = cloneObject(utxo);
+
     // make sure bip32 indices are unsigned
-    if (utxo.address_n && Array.isArray(utxo.address_n)) {
-        utxo.address_n = utxo.address_n.map(i => i >>> 0);
+    if (newUtxo.address_n && Array.isArray(newUtxo.address_n)) {
+        newUtxo.address_n = newUtxo.address_n.map(i => i >>> 0);
     }
-    // make sure that address_n is an array
-    if (utxo.address_n && typeof utxo.address_n === 'string') {
-        utxo.address_n = getHDPath(utxo.address_n);
+    // Make sure that address_n is an array
+    if (newUtxo.address_n && typeof newUtxo.address_n === 'string') {
+        newUtxo.address_n = getHDPath(newUtxo.address_n);
     }
 
-    return utxo as ProtoWithAddressN<T>;
+    return newUtxo as ProtoWithAddressN<T>;
```

Comment:
wow this commit looks unrelated. but I get the point - missing serialization. but shouldn't it be handled somewhere centrally?

Reply:
My thinking here was to have a pure function, since modifying the argument passed specially when it is an object in JS does not look like a good idea. If you modify the provided object in the function the side effects can be unpredictable. So it is safer to create a new object and return the new object modified.

Reply:
🤔 
btw: where it was failing? some CI test? Because I assume that it started failing because of missing serialization but in suite-desktop, connect is run inside node and I think there is no serialization either. So this should be happening there as well.

Reply:
It was failing in suite-web send test if I remember well. Basically it was modifying an object that was freezed, probably because it was an object from the reducer that is not allowed to be directly modified.

Reply:
Aha, I know.
Carlos, please lets do it like this:

- create an issue that there might be some places causing side/effects object mutations
- link this issue with a todo to this file (packages/connect/src/utils/pathUtils.ts)
- suite-destkop actually has serialization because data passed to connect calls first go through renderer-main interaface. 
- so lets mimic the same behaviour for connect-in-module for now. I would clone the input param of connect here https://github.com/trezor/trezor-suite/blob/feat/use-core-in-suite-web/packages/connect/src/impl/core-in-module.ts#L208

Reply:
Ok, I created issue https://github.com/trezor/trezor-suite/issues/15559 and liked it to that file.